utils/src/main/kotlin/Utils.kt

```diff
@@ -19,12 +19,24 @@
 
 package com.here.ort.utils
 
+import org.apache.logging.log4j.kotlin.KotlinLogger
+import org.apache.logging.log4j.kotlin.loggerOf
+
 import java.io.File
 import java.net.URI
 import java.net.URISyntaxException
 import java.security.Permission
 
-val log = org.apache.logging.log4j.kotlin.logger("ORT")
+/**
+ * Global map of loggers for classes so only one logger needs to be instantiated per class.
+ */
+val loggerOfClass = mutableMapOf<Any, KotlinLogger>()
+
+/**
+ * An extension property for adding a log instance to any (unique) class.
+ */
+val <reified T : Any> T.log: KotlinLogger
+    inline get() = loggerOfClass.getOrPut(T::class.java) { loggerOf(T::class.java) }
```

Comment:
Probably we wanna take care of thread safety here...

Reply:
Solved by using a `ConcurrentHashMap` instead.

Reply:
I just realized Log4j already uses a `LoggerRegistry` for logger "caching" internally itself, so I guess we do not need this at all, see https://github.com/apache/logging-log4j2/blob/log4j-2.12.0/log4j-api/src/main/java/org/apache/logging/log4j/spi/LoggerRegistry.java#L53

Reply:
Ok, then let's rely on this for now.

Reply:
> The performance impact is probably minimal, but still, can you think of any better solution except of the obvious to create the instances inside of the classes?

Unfortunately not.

> I just realized Log4j already uses a `LoggerRegistry` for logger "caching" internally itself

Again, unfortunately, this "caching" is only done foe the internal delegate object. So while no logger implementation is recreated, the `KotlinLogger` *is* recreated on each call to `log` unless we add your own "caching".

I currently don't really like any of the options 😞

Reply:
And about

> You have to use `putIfAbsent()` (or `computeIfAbsent()`) instead with `ConcurrentHashMap`

Koltin has a specialized implementation of `getOrPut()` for `ConcurrentHashMap` which already uses `putIfAbsent()`.